### PR TITLE
adding status updater / device plugin for kwok nodes

### PR DESCRIPTION
**Pull request by enoodle**

### CODE REVIEW

File: internal/kwok-gpu-device-plugin/controllers/configmap/controller.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		log.Fatalf("Failed to get cluster topology: %v", err)
+	}
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		kubeClient, 0, informers.WithNamespace(namespace))
+	c := &ConfigMapController{
+		kubeClient:      kubeClient,
+		cmInformer:      informerFactory.Core().V1().ConfigMaps().Informer(),
+		nodeLister:      informerFactory.Core().V1().Nodes().Lister(),
```
Comment (gshaibi) [inline internal/kwok-gpu-device-plugin/controllers/configmap/controller.go:46]: Do we still need the node lister?

File: internal/kwok-gpu-device-plugin/controllers/configmap/controller.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	listersv1 "k8s.io/client-go/listers/core/v1"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type ConfigMapController struct {
+	kubeClient      kubernetes.Interface
+	cmInformer      cache.SharedIndexInformer
```
Comment (gshaibi) [inline internal/kwok-gpu-device-plugin/controllers/configmap/controller.go:23]: Do we need to keep it in the struct?

File: internal/kwok-gpu-device-plugin/controllers/configmap/controller.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	nodeLister      listersv1.NodeLister
+	informerFactory informers.SharedInformerFactory
+	handler         cmhandler.Interface
+
+	clusterTopology *topology.ClusterTopology
```
Comment (gshaibi) [inline internal/kwok-gpu-device-plugin/controllers/configmap/controller.go:28]: Do we need to keep it in the struct?

**Review by gshaibi — APPROVED** [review]
Amazing. Thank you!

File: internal/kwok-gpu-device-plugin/controllers/configmap/controller.go (hunk 2 above)
Comment (enoodle) [reply internal/kwok-gpu-device-plugin/controllers/configmap/controller.go:23]: No, I will remove this

File: internal/kwok-gpu-device-plugin/controllers/configmap/controller.go (hunk 3 above)
Comment (enoodle) [reply internal/kwok-gpu-device-plugin/controllers/configmap/controller.go:28]: No, I will remove this

**Pull request merged**